src/benchmark/join_order_benchmark.cpp

```diff
@@ -21,6 +21,91 @@
+
+  const auto& complete_cast_table = table_info_by_name.at("complete_cast").table;
+  complete_cast_table->add_soft_key_constraint(
+      {{complete_cast_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& info_type_table = table_info_by_name.at("info_type").table;
+  info_type_table->add_soft_key_constraint(
+      {{info_type_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& keyword_table = table_info_by_name.at("keyword").table;
+  keyword_table->add_soft_key_constraint({{keyword_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& kind_type_table = table_info_by_name.at("kind_type").table;
+  kind_type_table->add_soft_key_constraint(
+      {{kind_type_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& link_type_table = table_info_by_name.at("link_type").table;
+  link_type_table->add_soft_key_constraint(
+      {{link_type_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& movie_companies_table = table_info_by_name.at("movie_companies").table;
+  movie_companies_table->add_soft_key_constraint(
+      {{movie_companies_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& movie_info_table = table_info_by_name.at("movie_info").table;
+  movie_info_table->add_soft_key_constraint(
+      {{movie_info_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& movie_info_idx_table = table_info_by_name.at("movie_info_idx").table;
+  movie_info_idx_table->add_soft_key_constraint(
+      {{movie_info_idx_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& movie_keyword_table = table_info_by_name.at("movie_keyword").table;
+  movie_keyword_table->add_soft_key_constraint(
+      {{movie_keyword_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& movie_link_table = table_info_by_name.at("movie_link").table;
+  movie_link_table->add_soft_key_constraint(
+      {{movie_link_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& name_table = table_info_by_name.at("name").table;
+  name_table->add_soft_key_constraint({{name_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& person_info_table = table_info_by_name.at("person_info").table;
+  person_info_table->add_soft_key_constraint(
+      {{person_info_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& role_type_table = table_info_by_name.at("role_type").table;
+  role_type_table->add_soft_key_constraint(
+      {{role_type_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
+
+  const auto& title_table = table_info_by_name.at("title").table;
+  title_table->add_soft_key_constraint({{title_table->column_id_by_name("id")}, KeyConstraintType::PRIMARY_KEY});
```

Comment:
If I look at these lines. Is there a reason not to loop over table_info_by_name and set the primary key on `id`?
Would have solved the `name` issue as well.

Reply:
Yeah, the FK issue you both mentioned makes sense to me.